### ISSUE TITLE
Coupon usage limits not honored 

### ISSUE THREAD

**Issue by ioweb-gr** [magento/magento2#38318]
### Preconditions and environment

-  Magento ver. 2.4.6

### Steps to reproduce

1. Create a cart rule with a specific coupon set to use only once 

<details>

![image](https://github.com/magento/magento2/assets/20220341/7c3efd00-0586-4fdd-a3e1-bfa4af78e19d)

</details>

2. Place an order with the coupon 

<details>

![image](https://github.com/magento/magento2/assets/20220341/3e53a3c1-49d4-4aab-8d4c-7580246b5039)

</details>

3. Try to place an order again and apply the coupon

<details>

![image](https://github.com/magento/magento2/assets/20220341/c7be071c-5beb-42a1-95ad-5b636f669278)

</details>

### Expected result

The coupon should not be applied as it's already used

### Actual result

The coupon is applied

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @ioweb-gr,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.

**Comment by engcom-Bravo** [comment]
Hi @ioweb-gr,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.


**Steps to reproduce**

- Create a cart rule with a specific coupon set to use only once
- Place an order with the coupon
- Try to place an order again and apply the coupon


![Test-Promotions-Marketing-Magento-Admin](https://github.com/magento/magento2/assets/51680745/a5b1b87a-8366-4050-a596-b916aca32e6c)


![-000000006-Orders-Operations-Sales-Magento-Admin (1)](https://github.com/magento/magento2/assets/51680745/5a1e024b-8327-4636-bf97-09ddab285d6b)


![Screenshot from 2024-01-02 12-19-47](https://github.com/magento/magento2/assets/51680745/d5d793bc-793e-43be-9239-5a261c747e5f)


We have updated coupon usage with this command `bin/magento queue:consumers:start sales.rule.update.coupon.usage` as per this document https://experienceleague.adobe.com/docs/commerce-operations/configuration-guide/cli/start-message-queues.html.

After updating coupon usage:

We are not able to apply the same coupon for the order.

Kindly recheck the behaviour on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.

Thanks.
